quicklogic/qlf_k4n8/utils/create_lib.py

```diff
@@ -0,0 +1,416 @@
+#!/usr/bin/env python3
+"""
+Creates a device library file
+"""
+import argparse
+import sys
+import csv
+import os
+from collections import namedtuple
+from datetime import date
+import lxml.etree as ET
+
+# =============================================================================
+
+# =============================================================================
+"""
+Pin properties
+name - pin_name
+dir - direction (input/output)
+used - specify 'yes' if user is using the pin else specify 'no'
+clk - specify associate clock
+"""
+PinData = namedtuple("PinData", "name dir used clk")
+
+# =============================================================================
+
+
+def main():
+    """
+    Creates a device library file by getting data from given csv and xml file
+    """
+    parser = argparse.ArgumentParser(
+        description='Creates a device library file.'
+    )
+    parser.add_argument(
+        "--lib",
+        "-l",
+        "-L",
+        type=str,
+        default="qlf_k4n8.lib",
+        help='The output device lib file'
+    )
+    parser.add_argument(
+        "--lib_name",
+        "-n",
+        "-N",
+        type=str,
+        required=True,
+        help='Specify library name'
+    )
+    parser.add_argument(
+        "--cell_name",
+        "-m",
+        "-M",
+        type=str,
+        required=True,
+        help='Specify cell name'
+    )
+    parser.add_argument(
+        "--csv",
+        "-c",
+        "-C",
+        type=str,
+        required=True,
+        help='Input pin-map csv file'
+    )
+    parser.add_argument(
+        "--xml",
+        "-x",
+        "-X",
+        type=str,
+        required=True,
+        help='Input interface-mapping xml file'
+    )
+
```

Comment:
The files `dedicated_pin_lib_data.txt` and `lib_header_template.txt` have hard-coded names and paths relative to this script. I presume that they are specific to a device. It may be better to add eg. an option to specify a path to the device folder where the files are. This would make the script more generic and usable for future devices to come.

Reply:
Incorporated this in the last checkin. Added a command line option to get template data folder path.